packages/babel-plugin-transform-es2015-block-scoping/src/index.js

```diff
@@ -488,17 +488,31 @@ class BlockScoping {
       }
     }
 
+    const addDeclarationsFromChild = (path, node) => {
+      node = node || path.node;
+      if (t.isClassDeclaration(node) || t.isFunctionDeclaration(node) || isBlockScoped(node)) {
+        if (isBlockScoped(node)) {
+          convertBlockScopedToVar(path, node, block, this.scope);
+        }
+        declarators = declarators.concat(node.declarations || node);
```

Comment:
If we are already deduplicating code, then we maybe can also deduplicate this `if`? as it basically twice in here I think.